cmd/kubeadm/app/apis/kubeadm/v1alpha1/types.go

```diff
@@ -32,6 +33,11 @@ type MasterConfiguration struct {
 type API struct {
 	AdvertiseAddresses []string `json:"advertiseAddresses"`
 	ExternalDNSNames   []string `json:"externalDNSNames"`
+	BindPort           int32    `json:"bindPort"`
+}
+
+type Discovery struct {
+	BindPort int32 `json:"bindPort"`
```

Comment:
It seems like these fields could be factored in a way to reduce duplication between NodeConfiguration and MasterConfiguration. It seems like API should be a field of NodeConfiguration and the bind port of the discovery api should be contained in API.